tipsy/TipViewController.swift

```diff
@@ -10,35 +10,79 @@ import UIKit
 
 class ViewController: UIViewController {
 
-    @IBOutlet weak var tipLabel: UILabel!
+    @IBOutlet private weak var tipLabel: UILabel!
     
-    @IBOutlet weak var totalLabel: UILabel!
+    @IBOutlet private weak var totalLabel: UILabel!
     
-    @IBOutlet weak var billField: UITextField!
-    @IBOutlet weak var tipControl: UISegmentedControl!
+    @IBOutlet private weak var billField: UITextField!
+    @IBOutlet private weak var tipControl: UISegmentedControl!
     
     private var bill = 0.0
+    private let defaults = UserDefaults.standard
+    
+    // MARK: - Lifecycle Methods
     
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view, typically from a nib.
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        // Get the last time the view closed
+        let timeAppClosed = defaults.object(forKey: UserDefaultKeys.timeAppClosed) as? Int ?? 0
+        // Set the default tip option if one exists, else default to 18%
+        if let defaultTipIndex = defaults.object(forKey: UserDefaultKeys.defaultTipIndex) as? Int {
+            tipControl.selectedSegmentIndex = defaultTipIndex
+            updateTip()
+        }
+        // Get the current time
+        let currentTime = Int(NSDate().timeIntervalSince1970)
+        // Check if enough time has passed to update the view
+        if currentTime - timeAppClosed < 600,
+            let billLastClosed = defaults.object(forKey: UserDefaultKeys.billLastClosed) as? String,
+            let percentageLastClosed = defaults.object(forKey: UserDefaultKeys.percentageLastClosed) as? Int {
```

Comment:
Also you can chain together `if` statements. I use these a lot of consolidate code.